sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosTable.scala

```diff
@@ -70,18 +71,12 @@ class CosmosTable(val transforms: Array[Transform],
   // database and table name from catalog, or the location of files for this table.
   override def name(): String = "com.azure.cosmos.spark.write"
 
-  /**
-    * Returns the schema of this table. If the table is not readable and doesn't have a schema, an
-    * empty schema can be returned here.
-    */
   override def schema(): StructType = {
-    // TODO: moderakh add support for schema inference
-    // for now schema is hard coded to make TestE2EMain to work
-    val hardCodedSchema = StructType(Seq(StructField("number", IntegerType), StructField("word", StringType)))
-    userProvidedSchema.getOrElse(hardCodedSchema)
+    userProvidedSchema.getOrElse(CosmosTableSchemaInferer.inferSchema(client, effectiveUserConfig))
   }
 
   override def capabilities(): util.Set[TableCapability] = Set(
+    TableCapability.ACCEPT_ANY_SCHEMA,
```

Comment:
Please add a brief comment why this ended-up being needed .

Reply:
Done